packages/models-library/src/models_library/api_schemas_webserver/functions_wb_schema.py

```diff
@@ -0,0 +1,290 @@
+    pass
+
+
+SolverJobID: TypeAlias = UUID
+
+
+class SolverFunction(FunctionBase):
+    function_class: Literal[FunctionClass.solver] = FunctionClass.solver
+    solver_key: ServiceKey
+    solver_version: ServiceVersion
+
+
+class RegisteredSolverFunction(SolverFunction, RegisteredFunctionBase):
+    pass
+
+
+class PythonCodeFunction(FunctionBase):
+    function_class: Literal[FunctionClass.python_code] = FunctionClass.python_code
+    code_url: str
+
+
+class RegisteredPythonCodeFunction(PythonCodeFunction, RegisteredFunctionBase):
+    pass
+
+
+Function: TypeAlias = Annotated[
+    ProjectFunction | PythonCodeFunction | SolverFunction,
+    Field(discriminator="function_class"),
+]
+RegisteredFunction: TypeAlias = Annotated[
+    RegisteredProjectFunction | RegisteredPythonCodeFunction | RegisteredSolverFunction,
+    Field(discriminator="function_class"),
+]
+
+FunctionJobCollectionID: TypeAlias = projects.ProjectID
+
+
+class FunctionJobBase(BaseModel):
+    title: str = ""
+    description: str = ""
+    function_uid: FunctionID
+    inputs: FunctionInputs
+    outputs: FunctionOutputs
+    function_class: FunctionClass
+
+
+class RegisteredFunctionJobBase(FunctionJobBase):
+    uid: FunctionJobID
+
+
+class ProjectFunctionJob(FunctionJobBase):
+    function_class: Literal[FunctionClass.project] = FunctionClass.project
+    project_job_id: ProjectID
+
+
+class RegisteredProjectFunctionJob(ProjectFunctionJob, RegisteredFunctionJobBase):
+    pass
+
+
+class SolverFunctionJob(FunctionJobBase):
+    function_class: Literal[FunctionClass.solver] = FunctionClass.solver
+    solver_job_id: ProjectID
+
+
+class RegisteredSolverFunctionJob(SolverFunctionJob, RegisteredFunctionJobBase):
+    pass
+
+
+class PythonCodeFunctionJob(FunctionJobBase):
+    function_class: Literal[FunctionClass.python_code] = FunctionClass.python_code
+
+
+class RegisteredPythonCodeFunctionJob(PythonCodeFunctionJob, RegisteredFunctionJobBase):
+    pass
+
+
+FunctionJob: TypeAlias = Annotated[
+    ProjectFunctionJob | PythonCodeFunctionJob | SolverFunctionJob,
+    Field(discriminator="function_class"),
+]
+
+RegisteredFunctionJob: TypeAlias = Annotated[
+    RegisteredProjectFunctionJob
+    | RegisteredPythonCodeFunctionJob
+    | RegisteredSolverFunctionJob,
+    Field(discriminator="function_class"),
+]
+
+
+class FunctionJobStatus(BaseModel):
+    status: str
+
+
+class FunctionJobCollection(BaseModel):
+    """Model for a collection of function jobs"""
+
+    title: str = ""
+    description: str = ""
+    job_ids: list[FunctionJobID] = []
+
+
+class RegisteredFunctionJobCollection(FunctionJobCollection):
+    uid: FunctionJobCollectionID
+
+
+class FunctionJobCollectionStatus(BaseModel):
+    status: list[str]
+
+
+class FunctionBaseError(OsparcErrorMixin, Exception):
+    pass
+
+
+class FunctionIDNotFoundError(FunctionBaseError):
+    """Exception raised when a function is not found"""
```

Comment:
minor: I think it's also safe to drop all the comments

Reply:
Ok, done. Pity, I would have prefered comments on all classes ;-)